### PR TITLE
Use MozTools 4.0 and vendored autoconf

### PR DESCRIPTION
**Pull request by sagudev**

### CODE REVIEW

**Comment by mrobinson** [comment]
AFAIU, moztool is a zip file created by us, so what changes are in moztools 4.0?

**Comment by sagudev** [comment]
> AFAIU, moztools is a zip file created by us, so what changes are in moztools 4.0?

It's based on mozillabuild 4.0. Here is PR for it: https://github.com/servo/servo-build-deps/pull/2. My plan is to get moztools-4.0 in servo-build-deps repo, then switch PR to that one and then get this landed.

**Comment by mrobinson** [comment]
Thanks for the explanation. This sounds like a great plan.

**Comment by mrobinson** [comment]
I have uploaded moztools-4.0 to https://github.com/servo/servo-build-deps/releases.

File: README.md [hunk 1]
```diff
@@ -13,11 +13,11 @@ are in the [rust-mozjs directory][r-m].
 
 ## Linux
 
-Install Clang (at least version 3.9) and autoconf v 2.13, for example on a
+Install Clang (at least version 3.9), for example on a
 Debian-based Linux:
 
 ```sh
-sudo apt-get install clang-6.0 autoconf2.13
```
Comment (mrobinson) [inline README.md:20]: This might need to include `build-essential` for Linux...

Comment (sagudev) [reply README.md:20]: Also clang 6.0 might be to old now.

**Review by jdm — APPROVED** [review]
This looks like a nice simplification.

**Pull request merged**